bftengine/src/bftengine/ReplicaImp.hpp

```diff
@@ -118,6 +118,23 @@ class ReplicaImp : public InternalReplicaApi, public ReplicaForStateTransfer {
   std::map<uint64_t, std::pair<Time, ClientRequestMsg*>>
       requestsOfNonPrimary;  // used to retransmit client requests by a non primary replica
   size_t NonPrimaryCombinedReqSize = 1000;
+  //
+  const std::thread::id MAIN_THREAD_ID;
+  uint16_t activeExecutions_ = 0;                     // represents the number of running executions at this moment.
+  std::deque<ClientRequestMsg*> deferredRORequests_;  // if we have active executions and the main thread fetches RO
+                                                      // request we will defer the message and store it in this queue,
+                                                      // we will handle this requests after the post execution finishes
```

Comment:
Please explain why the queues should not be thread-safe.

Reply:
Done thanks.